src/main/kotlin/io/bazel/kotlin/builder/tasks/jvm/compilation_task.kt

```diff
@@ -170,6 +170,47 @@ internal fun JvmCompilationTask.kaptArgs(
   }
 }
 
+internal fun JvmCompilationTask.kspArgs(
+  plugins: InternalCompilerPlugins
+): CompilationArgs {
+  return CompilationArgs().apply {
+    plugin(plugins.kspSymbolProcessingCommandLine)
+    plugin(plugins.kspSymbolProcessingApi) {
+      flag("-Xallow-no-source-files")
+
+      val values = arrayOf(
+        "apclasspath" to listOf(inputs.processorpathsList.joinToString(File.pathSeparator)),
+        // projectBaseDir shouldn't matter because incremental is disabled
+        "projectBaseDir" to listOf(directories.incrementalData),
+        // Disable incremental mode
+        "incremental" to listOf("false"),
+        // Directory where class files are written to. Files written to this directory are class
+        // files being written directly from the annotation processor, not Kotlinc
+        "classOutputDir" to listOf(directories.generatedClasses),
+        // Directory where generated Java sources files are written to
+        "javaOutputDir" to listOf(directories.generatedJavaSources),
+        // Directory where generated Kotlin sources files are written to
+        "kotlinOutputDir" to listOf(directories.generatedSources),
```

Comment:
`kt_compiler_plugin` provide the following template variables:
    - `{generatedSources}` template provides a generated source folder
    - `{generatedClasses}` for class generation via the `kt_compiler_plugin`
    - `{temp}` for incremental work

`kt_compiler_plugin.stubs_phase` runs before compilation and kapt. We might consider renaming this to `header`/`abi` phase depending on how we structure the actions.

 Do we need another template variable for java sources to be generated? If so, should it be propagated to the java compile?

ref: https://github.com/bazelbuild/rules_kotlin/blob/master/docs/kotlin.md#kt_compiler_plugin